File: 162-find-peak-element/find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int i = 0, j = nums.size()-1;
        while(i <= j){
            int mid = midpoint(i,j);

            if (mid > 0 && nums[mid] < nums[mid-1]) j = mid - 1;
            else if (mid < nums.size()-1 && nums[mid] < nums[mid+1]) i = mid + 1;
            else return mid;
        }

        return -1;
    }
};

File: 162-find-peak-element/find-peak-element.go
func findPeakElement(nums []int) int {
    left, right := 0, len(nums)-1

    for left <= right {
        mid := left + (right - left)/2

        if mid > 0 && nums[mid] < nums[mid-1] {
            right = mid - 1
        } else if mid < len(nums)-1 && nums[mid] < nums[mid+1] {
            left = mid + 1
        } else {
            return mid
        }
    }

    return -1
}